File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        
        Set<Integer> set = new HashSet<>();

        for (int i = 0; i < nums.length; i++){
            set.add(nums[i]);
        }
        
        int res = 0;
        for (int i = 0; i < nums.length; i++){
            if (!set.contains(nums[i]-1)){
                int current = nums[i];
                int max = 0;
                while (set.contains(current + 1)){
                    max++;
                    current++;
                }
                res = Math.max(res, max+1);
            }   
        }
        
        return res;
    }
}

File: 128-longest-consecutive-sequence/128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        seq = {}
        for i in nums:
            seq[i] = i + 1
        
        dp = {}
        ans = 0
        for j in seq:
            count = 0
            x = j
            while seq[x] in seq:
                if x in dp:
                    count += dp[x]
                    break
                else:
                    count += 1
                    x += 1
            
            dp[j] = count
            ans = max(ans, count + 1)
        # print(dp)
            
        return ans
                